src/Compilers/Core/Portable/Symbols/TypeSymbolEqualityComparer.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// TODO
+    /// </summary>
+    public sealed class TypeSymbolEqualityComparer : IEqualityComparer<ITypeSymbol>
+    {
+        /// <summary>
+        /// Compares two types based on the default comparison rules, equivalent to calling <see cref="IEquatable{ISymbol}.Equals(ISymbol)"/>
+        /// </summary>
+        public static TypeSymbolEqualityComparer Default = new TypeSymbolEqualityComparer(TypeCompareKind.AllNullableIgnoreOptions);
+
+        /// <summary>
+        /// Compares two types considering everything available to the compiler
```

Comment:
reading this as an external user, i would have no ideat wahat "everything available to the compiler" means.